input/postgres/statements.go

```diff
@@ -247,7 +248,13 @@ func GetStatements(ctx context.Context, server *state.Server, logger *util.Logge
 		}
 
 		if showtext {
-			statementTexts[key] = receivedQuery.String
+			select {
+			// Since normalizing can take time, explicitly check for cancellations
+			case <-ctx.Done():
+				return nil, nil, nil, ctx.Err()
+			default:
+				fingerprintAndNormalize(key, receivedQuery.String, server, &statements, &statementTextsByFp)
```

Comment:
I think you don't need to pass the pointer of the map here as the map is already a reference. It should simplify the `fingerprintAndNormalize` implementation too.

Reply:
> `statements` and `statementTextsByFp` are not currently references, since they're defined in this function. 

Uhm, I don't think it's true. Or rather, even if it's defined (initialized), it's still a reference (well, or pointer?).
https://groups.google.com/g/golang-nuts/c/xzdPCjKORNA/m/7qd0BEklVqAJ

Here is super simple snippet demonstrates that:
https://go.dev/play/p/5xgmqAV5Ier

Reply:
Hmm, that's an odd language design (that seems to be poorly-documented), but I updated the code and added a logger statement at the end of `GetStatements` to confirm the map is actually populated, and it looks good. I'll update this PR.